### DISCUSSION TITLE
OOP-IEC61131-3--Curso-Youtube/es/Clases%20y%20Objetos/metodo/

### DISCUSSION THREAD

**Discussion by giscus** [runtimevic/OOP-IEC61131-3--Curso-Youtube#16] [General]
# OOP-IEC61131-3--Curso-Youtube/es/Clases%20y%20Objetos/metodo/

Curso OOP 61131-3 PLC

https://runtimevic.github.io/OOP-IEC61131-3--Curso-Youtube/es/Clases%20y%20Objetos/metodo/

**Comment by AndresFRodriguezC** [comment]
Hola, excelente todo el trabajo que has hecho y como me ha ayudado. Quisiera preguntarle en que casos se debe hacer uso de una accion en codesys o un metodo? y otra duda, en una propiedad comprendo lo del set y el get pero yo puedo anadir codigo de control o simplemente siempre son los igualadores.

Gracias.

**Reply by runtimevic** [reply]
Hola Andres, gracias por reconocer este trabajo...
- A una acción le puedes dar por ejemplo la utilidad  de en un Bloque ya sea una POU,FB, FC dividirla en varias partes y esto lo puedes conseguir con diferentes llamadas a acciones.., de esta forma tu Bloque quedaria mas separado y seria mas legible al tener diferentes partes diferenciadas...
- El uso de un método es mas para la programación OOP en el cual por ejemplo un FB (Clase) el método seria para encapsular un comportamiento de una entidad.
- en la propiedades no son solo igualadores para poder leer/escribrir variables de tu FB (clase), si no como bien dices también los puedes utilizar para añadir cualquier tipo de codigo de control, etc... que necesites.
Espero que sigas estudiando y aplicando OOP en el mundo de los PLCs ya que creo que puede aportar muchas ventajas si se saben aplicar adecuadamente...
